components/net/fetch/cors_cache.rs

```diff
@@ -122,21 +116,21 @@ impl CORSCache {
 
     /// Returns true if an entry with a
     /// [matching header](https://fetch.spec.whatwg.org/#concept-cache-match-header) is found
-    pub fn match_header(&mut self, request: CacheRequestDetails, header_name: &str) -> bool {
+    pub fn match_header(&mut self, request: Request, header_name: &str) -> bool {
```

Comment:
Why is this changing the ownership of request? Can it not use a reference instead?